### PR TITLE
[sp] add text column stats

### PR DESCRIPTION
**Pull request by shrey-mage**
# Summary
- add word distribution chart to column-specific "Visualizations" tab
- display min/mean/max word counts in column-specific "Reports" tab
- display min/mean/max word counts in feature profiles

# Tests
image
image
image

cc: @johnson-mage @wangxiaoyou1993 @dy46

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Can we also min/max/mean string length in the statistics?

**Comment by johnson-mage** [comment]
wondering if there's a way to indicate that "min/max/mean" for a text column is word count in the feature profiles table. May not be clear to a user what "min" for a text col is.

**Comment by shrey-mage** [comment]
> wondering if there's a way to indicate that "min/max/mean" for a text column is word count in the feature profiles table. May not be clear to a user what "min" for a text col is.

agreed, was thinking about this. Maybe we can put a label on hover? Maybe Kai's tooltip explanation design?

**Comment by wangxiaoyou1993** [comment]
Or we should just use different stats name in features profiles. Something like this
image

File: mage_ai/frontend/components/datasets/columns/ColumnReports.tsx [hunk 1]
```diff
@@ -101,35 +107,69 @@ function ColumnReports({
...
+        value: isNumeric(median) ? roundNumber(median) : median,
+      },
+      {
+        name: 'Average value',
+        value: isNumeric(average) ? roundNumber(average) : average,
+      },
+    );
+  }
+
+  if (columnType === ColumnTypeEnum.TEXT) {
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/columns/ColumnReports.tsx:144]: do we also need to check for other string column types like CATEGORY?

File: mage_ai/frontend/components/datasets/overview/index.tsx [hunk 2]
```diff
@@ -317,7 +318,13 @@ function DatasetOverview({
               column={selectedColumn}
               features={features}
               insights={insightsOverview}
-              statisticsByColumn={statistics?.[`${selectedColumn}/value_counts`] || {}}
+              statisticsByColumn={
+                (colType === ColumnTypeEnum.TEXT 
+                  ?
+                  statistics?.[`${selectedColumn}/word_distribution`]
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/overview/index.tsx:324]: can put on line above

File: mage_ai/frontend/components/datasets/overview/index.tsx [hunk 3]
```diff
@@ -317,7 +318,13 @@ function DatasetOverview({
...
+                  :
+                  statistics?.[`${selectedColumn}/value_counts`]
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/overview/index.tsx:326]: this too

**Review by johnson-mage — APPROVED** [review]

**Pull request merged**